cumulus/parachains/runtimes/assets/common/src/matching.rs

```diff
@@ -14,68 +14,116 @@
-	prelude::*,
+use frame_support::{
+	pallet_prelude::Get,
+	traits::{Contains, ContainsPair},
 };
+use xcm::prelude::*;
+
+pub struct StartsWith<T>(sp_std::marker::PhantomData<T>);
+impl<LocationValue: Get<Location>> Contains<Location> for StartsWith<LocationValue> {
+	fn contains(t: &Location) -> bool {
+		t.starts_with(&LocationValue::get())
+	}
+}
+
+pub struct Equals<T>(sp_std::marker::PhantomData<T>);
+impl<LocationValue: Get<Location>> Contains<Location> for Equals<LocationValue> {
+	fn contains(t: &Location) -> bool {
+		t == &LocationValue::get()
+	}
+}
+
+pub struct StartsWithExplicitGlobalConsensus<T>(sp_std::marker::PhantomData<T>);
+impl<Network: Get<NetworkId>> Contains<Location> for StartsWithExplicitGlobalConsensus<Network> {
+	fn contains(t: &Location) -> bool {
+		matches!(t.interior.global_consensus(), Ok(requested_network) if requested_network.eq(&Network::get()))
+	}
+}
 use xcm_builder::ensure_is_remote;
 
 frame_support::parameter_types! {
-	pub LocalMultiLocationPattern: MultiLocation = MultiLocation::new(0, Here);
-	pub ParentLocation: MultiLocation = MultiLocation::parent();
+	pub LocalLocationPattern: Location = Location::new(0, Here);
+	pub ParentLocation: Location = Location::parent();
 }
 
 /// Accepts an asset if it is from the origin.
 pub struct IsForeignConcreteAsset<IsForeign>(sp_std::marker::PhantomData<IsForeign>);
-impl<IsForeign: ContainsPair<MultiLocation, MultiLocation>> ContainsPair<MultiAsset, MultiLocation>
+impl<IsForeign: ContainsPair<Location, Location>> ContainsPair<Asset, Location>
 	for IsForeignConcreteAsset<IsForeign>
 {
-	fn contains(asset: &MultiAsset, origin: &MultiLocation) -> bool {
+	fn contains(asset: &Asset, origin: &Location) -> bool {
 		log::trace!(target: "xcm::contains", "IsForeignConcreteAsset asset: {:?}, origin: {:?}", asset, origin);
-		matches!(asset.id, Concrete(ref id) if IsForeign::contains(id, origin))
+		matches!(asset.id, AssetId(ref id) if IsForeign::contains(id, origin))
 	}
 }
 
-/// Checks if `a` is from sibling location `b`. Checks that `MultiLocation-a` starts with
-/// `MultiLocation-b`, and that the `ParaId` of `b` is not equal to `a`.
-pub struct FromSiblingParachain<SelfParaId>(sp_std::marker::PhantomData<SelfParaId>);
-impl<SelfParaId: Get<ParaId>> ContainsPair<MultiLocation, MultiLocation>
-	for FromSiblingParachain<SelfParaId>
+/// Checks if `a` is from sibling location `b`. Checks that `Location-a` starts with
+/// `Location-b`, and that the `ParaId` of `b` is not equal to `a`.
+pub struct FromSiblingParachain<SelfParaId, L = Location>(
+	sp_std::marker::PhantomData<(SelfParaId, L)>,
+);
+impl<SelfParaId: Get<ParaId>, L: TryFrom<Location> + TryInto<Location> + Clone> ContainsPair<L, L>
+	for FromSiblingParachain<SelfParaId, L>
 {
-	fn contains(&a: &MultiLocation, b: &MultiLocation) -> bool {
+	fn contains(a: &L, b: &L) -> bool {
+		let a: Location = if let Ok(location) = (*a).clone().try_into() {
+			location
+		} else {
+			return false;
+		};
```

Comment:
@franciscoaguirre I see this pattern on several places here and also https://github.com/paritytech/polkadot-sdk/pull/1230/files#diff-cab606431ef00b3e6b8ad56805677b3de95416ad97232d26baacfa1fadd867a6R28-R37,
just curious if it could be simplified somehow, e.g. with `let Ok() = .. else { return false}` or `.ok_or(..)`, 
or maybe with `match`, something like:
```
let (a, b) = match ((*a).clone().try_into(), (*n).clone().try_into()) {
    (Ok(a), Ok(b)) if a.starts_with(&b)  => (a, b)
    _ => return false
}
```